calico_containers/adapter/datastore.py

```diff
@@ -658,38 +700,72 @@ def profile_update_rules(self, profile):
         self.etcd_client.write(rules_path, profile.rules.to_json())
 
     @handle_errors
-    def add_workload_to_profile(self, hostname, profile_name, container_id):
+    def append_profiles_to_endpoint(self, endpoint_id, profile_names):
```

Comment:
these methods need to take the full set of IDs and handle there being multiple endpoints (error)